### PR TITLE
Support configuring multiple SSH ports.

### PR DESCRIPTION
**Pull request by bfrieds**
# Testing
Set up sshd on port 12346 for brian-hvd-2 and 12345 for brian-hvd-3, ran the below command on `brian-hvd-1` and verified it ran to completion.
``` 
NCCL_DEBUG=INFO horovodrun -np 6 -H brian-hvd-2:2,brian-hvd-1:2,brian-hvd-3:2 --ssh-ports 12346,12344,12345
--verbose python code/source/horovod/examples/tensorflow_keras_mnist.py
```

### CODE REVIEW

File: horovod/run/run.py [hunk 1]
```diff
@@ -744,6 +773,8 @@ def _run(args):
             params += str(args.hosts) + ' '
         if args.ssh_port:
             params += str(args.ssh_port)
+        elif args.ssh_port:
```
Comment (aaron276h) [inline horovod/run/run.py]: Should this be `args.ssh_ports`?

File: horovod/run/run.py [hunk 2]
```diff
@@ -410,6 +434,9 @@ def parse_args():
     parser.add_argument('-p', '--ssh-port', action='store', dest='ssh_port',
                         type=int, help='SSH port on all the hosts.')
 
+    parser.add_argument('--ssh-ports', action='store', dest='ssh_ports',
```
Comment (aaron276h) [inline horovod/run/run.py]: Maybe assert that both `--ssh-ports` and `-ssh-port` aren't set at the same time?

**Comment by bfrieds** [comment]
> This looks really good, one question I had about how you tested it, did you disable SSH on all but the available ports? (e.g., was 12344 the only open ssh port on brian-hvd-1)

Yeah I verified it failed when sshd was not running on that port!

**Comment by aaron276h** [comment]
Yeah the chief machine doesn't use ssh. I had assumed `brian-hvd-2` was your chief because it was first in the list.

**Comment by bfrieds** [comment]
> I had assumed brian-hvd-2 was your chief because it was first in the list.

Yeah I wanted to verify this worked without having the first one be the chief :)

**Comment by aaron276h** [comment]
@bfrieds Yeah it does not use `ssh` for the chief, I had just assumed `brian-hvd-2` was your chief because it was the first in the list.

**Pull request merged**